### DISCUSSION TITLE
ReactNode和GNode的区别是什么呢？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [antvis/G6#6549] [Q&A] [answered]
@antv/g6-extension-react 提供了适用于 React 框架的节点：

1.ReactNode React 节点
2.GNode 采用 JSX 写法的 @antv/g 节点

请问这两种节点有什么区别？它们各自使用的场景是什么呢？

**Comment by Rey-Wang** [comment]
按照我的理解，reactNode 就是直接支持所有 ReactNode，GNode 相当于 @antv/g 预设了一些 ReactNode，方便快速生成带样式布局的节点

**Comment by Aarebecca** [comment, marked as answer]
是的，ReactNode 支持原生的 React 组件作为节点，GNode 是使用 JSX 的写法，但是只能使用特定的节点类型，如 Rect\Circle 等

**Reply by liaodalin19903** [reply]
也就是说，ReactNode的自定义范围更加广泛，GNode是预先定义好的有限的节点。
